pkg/util/filesystem/util_windows.go

```diff
@@ -260,3 +261,79 @@ func Chmod(path string, filemode os.FileMode) error {
 func IsAbs(path string) bool {
 	return filepath.IsAbs(path) || strings.HasPrefix(path, `\`) || strings.HasPrefix(path, `/`)
 }
+
+// IsPathValidForMount checks if the given `path` is a valid mount point.
+// Return whether the path is valid and error encountered if any
+func IsPathValidForMount(path string) (bool, error) {
+	path = strings.TrimSuffix(path, string(filepath.Separator))
+	dir, err := os.Lstat(path)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return false, err
+		}
+
+		return false, nil
+	}
+
+	if dir.IsDir() {
+		// the path may not be mounted yet
+		return true, nil
+	}
+
+	// for windows NTFS, check if the path is symlink instead of directory.
+	isSymlink := dir.Mode()&os.ModeSymlink != 0 || dir.Mode()&os.ModeIrregular != 0
+
+	// mounted folder created by SetVolumeMountPoint may still report ModeSymlink == 0
+	mountedFolder, err := IsMountedFolder(path)
```

Comment:
if it's a mounted folder (mounted with disk), `dir.Mode()&os.ModeIrregular != 0` should return true

Reply:
It becomes a bit complicated after rebasing as currently for Go 1.23+,
`dir.Mode()&os.ModeSymlink != 0 || dir.Mode()&os.ModeIrregular != 0`
should do the same as `IsMountedFolder` in most cases.
So I'm hesitated to change the condition from `isSymlink || mountedFolder` to `isSymlink && mountedFolder`.

Anyway, `IsMountedFolder` would be the most preferred way from the Microsoft doc regardless of the Go API change.